app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -58,22 +58,17 @@
 
     private final static String TAG = "SmsDetector";
 
-    private DataInputStream dis;
     private DataOutputStream dos;
+    BufferedReader in;
     private AimsicdService mAimsicdService;
-    private SharedPreferences prefs;
     private boolean mBound;
     private AIMSICDDbAdapter dbacess;
     private Context tContext;
-    private String[] SILENT_ONLY_TAGS;
+    private String[] LOADED_DETECTION_STRINGS;
+    private int B_LINED_SIZE = 30;// how many previous lines to hold
+    private String[] BUFFEREDLINES = new String[B_LINED_SIZE];//for holding previous lines from logcat that hold data we need
```

Comment:
This is another part of LQ code which shouldn't be merged, sorry. EVA said you don't have time for code review, I'm going to make code reviews here in my free time.